clients/src/main/java/org/apache/kafka/clients/consumer/internals/StreamsGroupHeartbeatRequestManager.java

```diff
@@ -237,6 +237,7 @@ private static List<StreamsGroupHeartbeatRequestData.TopicInfo> getRepartitionTo
                     repartitionTopicInfo.topicConfigs().add(new StreamsGroupHeartbeatRequestData.KeyValue().setKey(k).setValue(v))
                 );
                 repartitionTopicsInfo.add(repartitionTopicInfo);
+                repartitionTopicInfo.topicConfigs().sort(Comparator.comparing(StreamsGroupHeartbeatRequestData.KeyValue::key).thenComparing(StreamsGroupHeartbeatRequestData.KeyValue::value));
```

Comment:
Why do we need `thanComparing`? Should the `key` not be unique? I see that `topicConfigs()` returns `List<KeyValue>` (so maybe we don't have strict guarantee, and I am sure there is reasons why this is not a `Map<String, Object>` ?).

Reply:
You are right, since this is built from a map, we don't really need it unless the code somehow changes. removed.